src/MessageTrait.php

```diff
@@ -140,6 +140,9 @@ public function withBody(StreamInterface $body): self
     private function setHeaders(array $headers): void
     {
         foreach ($headers as $header => $value) {
+            if (\is_int($header)) {
```

Comment:
```suggestion
            // The array key could have been converted to an int. 
            if (\is_int($header)) {
```

Reply:
Squashed commits and added a comment here.